haskell-compile.el

```diff
@@ -143,19 +158,50 @@ node `(haskell-mode)compilation' for more details."
   (interactive "P")
   (save-some-buffers (not compilation-ask-about-save)
                          compilation-save-buffers-predicate)
-  (let* ((cabdir (haskell-cabal-find-dir))
-         (command1 (if (eq edit-command '-)
-                       haskell-compile-cabal-build-alt-command
-                     haskell-compile-cabal-build-command))
-         (srcname (buffer-file-name))
-         (command (if cabdir
-                      (format command1 cabdir)
-                    (if (and srcname (derived-mode-p 'haskell-mode))
-                        (format haskell-compile-command srcname)
-                      command1))))
-    (when (and edit-command (not (eq edit-command '-)))
-      (setq command (compilation-read-command command)))
+  (if-let ((stackdir (locate-dominating-file buffer-file-name "stack.yaml")))
+      (haskell-compile-stack stackdir edit-command)
+    (if-let ((cabaldir (haskell-cabal-find-dir)))
+        (haskell-compile-cabal cabaldir edit-command)
+      (let ((srcfile (buffer-file-name)))
+        (haskell-compile-ghc srcfile edit-command)))))
+
+(setq haskell-compile-stack-last nil)
+(defun haskell-compile-stack (dir edit-command)
+  (let* ((default (or haskell-compile-stack-last
+                      haskell-compile-stack-build-command))
```

Comment:
I can't find defs for `haskell-compile-build-command` or `haskell-compile-build-alt-command`.

Reply:
good catch...